### PR TITLE
added progress popup while starting up AtlasMap #71

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -87,8 +88,10 @@ function launchAtlasMapLocally(atlasmapExecutablePath: string, port: string): Pr
 					let text = dec.decode(data);
 					atlasMapServerOutputChannel.append(text);
 					if (text.indexOf("### AtlasMap Data Mapper UI started") > 0) {
+						atlasMapUIReady = true;
 						const url = "http://localhost:" + port;
 						openURL(url);
+						atlasMapUIReady = true;
```
Comment (apupier) [inline src/extension.ts]: why the atlasMapUIReady is set twice to true? at this line and 3 lines before.

Comment (lhein) [reply src/extension.ts]: nice catch.

**Comment by lhein** [comment]
> suggestions for later improvements:
    * close the "Starting AtlasMap instance at port XX" when the progress is finished or use only the progress bar and not provide the "Starting AtlasMap instance at port XX" information message
    * provide more information in the progress monitor , telling if it it will open in internal editor or in external default system browser

I agree with all of the above.  The pain point is that we cannot auto close the infomessages. See https://github.com/Microsoft/vscode/issues/2732. MS wants the info messages handled as some forced user interaction thing. If we don't want that they suggest using the setStatusMessage where a timeout can be provided. I am not really a fan of that either.

**Comment by apupier** [comment]
> I agree with all of the above. The pain point is that we cannot auto close the infomessages. See [Microsoft/vscode#2732](https://github.com/Microsoft/vscode/issues/2732). MS wants the info messages handled as some forced user interaction thing. If we don't want that they suggest using the setStatusMessage where a timeout can be provided. I am not really a fan of that either. Though the message closes at some point. It just takes some more time.

I created https://github.com/jboss-fuse/vscode-atlasmap/issues/90 to see later what we can improve

**Comment by apupier** [comment]
> I'm wondering what is happening if AtlasMap failed to start? Shoulnd't we add "atlasMapUIReady = true;" in the catch l.99?

I don't see this question answered. I missed it in the modified code? or the github interface?

**Comment by apupier** [comment]
> > I'm wondering what is happening if AtlasMap failed to start? Shoulnd't we add "atlasMapUIReady = true;" in the catch l.99?
> 
> I don't see this question answered. I missed it in the modified code? or the github interface?

ah, there is a timeout now, right?
https://github.com/jboss-fuse/vscode-atlasmap/pull/88/files#diff-45327f86d4438556066de133327f4ca2R144

**Comment by lhein** [comment]
Correct. I even mentioned it in the commit message.

**Pull request merged**